x-pack/test/integrations_manager_api_integration/apis/list.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import ServerMock from 'mock-http-server';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function({ getService }: FtrProviderContext) {
+  describe('list', () => {
+    const server = new ServerMock({ host: 'localhost', port: 6666 });
+    beforeEach(() => {
+      server.start(() => {});
+    });
+    afterEach(() => {
+      server.stop(() => {});
+    });
+    it('lists all packages from the registry', async () => {
+      const searchResponse = [
+        {
+          description: 'First integration package',
+          download: '/package/first-1.0.1.tar.gz',
+          name: 'first',
+          title: 'First',
+          type: 'integration',
+          version: '1.0.1',
+        },
+        {
+          description: 'Second integration package',
+          download: '/package/second-2.0.4.tar.gz',
+          icons: [
+            {
+              src: '/package/second-2.0.4/img/icon.svg',
+              type: 'image/svg+xml',
+            },
+          ],
+          name: 'second',
+          title: 'Second',
+          type: 'integration',
+          version: '2.0.4',
+        },
+      ];
+      server.on({
+        method: 'GET',
+        path: '/search',
+        reply: {
+          status: 200,
+          headers: { 'content-type': 'application/json' },
+          body: JSON.stringify(searchResponse),
+        },
+      });
+
+      const supertest = getService('supertest');
+      const fetchPackageList = async () => {
+        const response = await supertest
+          .get('/api/integrations_manager/list')
+          .set('kbn-xsrf', 'xxx')
+          .expect(200);
+        return response.body;
+      };
+
+      const listResponse = await fetchPackageList();
+
+      expect(listResponse.length).to.be(2);
+      expect(listResponse[0]).to.eql({ ...searchResponse[0], status: 'not_installed' });
+      expect(listResponse[1]).to.eql({ ...searchResponse[1], status: 'not_installed' });
```

Comment:
Another behavior we could test is that the [response is sorted (by name)](https://github.com/elastic/kibana/blob/b5b700c82fa8088509570e12e593217fd01c8fad/x-pack/legacy/plugins/integrations_manager/server/integrations/get.ts#L40-L47)

Not critical for this PR but a stable/sorted list is another behavior/characteristic of that function. The registry used to return results in a different order each call, so we sort before returning the results. The registry now has stable ordered results, but we still want to ensure that the results we return are sorted, regardless of what the registry returns.

Again, not critical for this PR.

Reply:
Good point, I've added a test for this.